abr_control/controllers/path_planners/second_order.py

```diff
@@ -52,12 +52,21 @@ def step(self, state, target):
             use the format state=np.array([posx, posy, posz, velx, vely, velz])
         target : numpy.array
             the target position of the system
+        w : float
+            the natural frequency
+        zeta : float
+            the damping ratio
+        threshold: float, Optional (Default: 0.02)
+            in the units of y and target, the distance from the target before
+            the filtered target becomes the target
         """
-        y = state[:3]
-        dy = state[3:]
-        ddy = self.w**2 * target - dy * self.zeta * self.w - y * self.w**2
-        state += np.hstack((dy, ddy)) * self.dt
-        return state
+        # check if within distance threshold
+        if np.linalg.norm(y-target) < threshold:
```

Comment:
what is this doing, again?

Reply:
Sets the filtered target to the final target position once within the threshold. It helps speed up movement a bit towards the end. Otherwise the last few cm can go slowly and the increased friction can lower performance.

Reply:
can you add that in a comment?

Reply:
It's already in the docstrings